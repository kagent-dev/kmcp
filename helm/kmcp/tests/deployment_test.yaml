suite: Test deployment template
templates:
  - deployment.yaml

tests:
  - it: should create deployment with default values
    template: deployment.yaml
    set:
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set custom replica count
    template: deployment.yaml
    set:
      controller.replicaCount: 3
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include metrics port when metrics enabled
    template: deployment.yaml
    set:
      controller.metrics.enabled: true
      controller.metrics.bindAddress: ":8443"
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include health probe ports when health probe enabled
    template: deployment.yaml
    set:
      controller.healthProbe.bindAddress: ":8081"
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set custom resources
    template: deployment.yaml
    set:
      resources:
        limits:
          cpu: 1000m
          memory: 256Mi
        requests:
          cpu: 100m
          memory: 128Mi
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include pod annotations when specified
    template: deployment.yaml
    set:
      podAnnotations:
        prometheus.io/scrape: "true"
        prometheus.io/port: "8443"
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include image pull secrets when specified
    template: deployment.yaml
    set:
      imagePullSecrets:
        - name: my-registry-secret
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include node selector when specified
    template: deployment.yaml
    set:
      nodeSelector:
        kubernetes.io/os: linux
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should include tolerations when specified
    template: deployment.yaml
    set:
      tolerations:
        - key: "key"
          operator: "Equal"
          value: "value"
          effect: "NoSchedule"
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set security context
    template: deployment.yaml
    set:
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
            - "ALL"
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set pod security context
    template: deployment.yaml
    set:
      podSecurityContext:
        runAsNonRoot: true
        seccompProfile:
          type: RuntimeDefault
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: should set termination grace period
    template: deployment.yaml
    set:
      image.repository: test-repo
      image.tag: v1.0.0
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {} 