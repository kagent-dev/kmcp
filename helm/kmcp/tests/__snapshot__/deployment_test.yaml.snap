should create deployment with default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should include health probe ports when health probe enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should include image pull secrets when specified:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          imagePullSecrets:
            - name: my-registry-secret
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should include metrics port when metrics enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should include node selector when specified:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          nodeSelector:
            kubernetes.io/os: linux
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should include pod annotations when specified:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
            prometheus.io/port: "8443"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should include tolerations when specified:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          tolerations:
            - effect: NoSchedule
              key: key
              operator: Equal
              value: value
          volumes: []
should set custom replica count:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 3
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should set custom resources:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 1000m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should set pod security context:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should set security context:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
should set termination grace period:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kmcp
        control-plane: controller-manager
        helm.sh/chart: kmcp-0.1.0
      name: RELEASE-NAME-controller-manager
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: kmcp
          control-plane: controller-manager
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: manager
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: kmcp
            control-plane: controller-manager
        spec:
          containers:
            - args:
                - --leader-elect
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8443
              command:
                - /manager
              image: test-repo:v1.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: health
                initialDelaySeconds: 15
                periodSeconds: 20
              name: manager
              ports:
                - containerPort: 8443
                  name: metrics
                  protocol: TCP
                - containerPort: 8081
                  name: health
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: health
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits:
                  cpu: 500m
                  memory: 128Mi
                requests:
                  cpu: 10m
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
              volumeMounts: []
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: RELEASE-NAME-kmcp-controller-manager
          terminationGracePeriodSeconds: 10
          volumes: []
